[3/N][BE][ST Deprecate] Remove Replicated Tensor Dependency

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95453
* #95450

Please use distributed tensor instead. We are deprecating ShardedTensor.

Commit:
[3/N] Remove Replicated Tensor

[ghstack-poisoned]

fduwjj commented:
@pytorchbot rebase

Commit:
Update on "[3/N][BE][ST Deprecate] Remove Replicated Tensor"


Please use distributed tensor instead. We are deprecating ShardedTensor.



[ghstack-poisoned]

wanchaol reviewed (approved):
lgtm, thanks for cleaning this up!

Commit:
Update on "[3/N][BE][ST Deprecate] Remove Replicated Tensor"


Please use distributed tensor instead. We are deprecating ShardedTensor.



[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge

wanchaol commented:
> > Please use distributed tensor instead. We are deprecating ShardedTensor.
> 
> @fduwjj Is this deprecation going to be part of the 2.0 release? Or left for 2.1?
> 
> I'm asking because in Lightning, we have this bit of code: https://github.com/Lightning-AI/lightning/blob/36c39ea70273aeff300a78f9fa4f129b72ee2bfb/src/lightning/pytorch/core/module.py#L1434-L1453
> 
> That we would want to skip based on the torch version, if ShardedTensor is deprecated.

@carmocca I think we are going to have deprecation msg for ShardedTensor in PyTorch 2.1, and will attempt to remove it in PyTorch 2.2, the warning message should fade away after https://github.com/pytorch/pytorch/pull/95631 (for 2.0 official release package it should not have the warnings I believe)
